tests/test_device_tools.py

```diff
@@ -116,3 +116,17 @@ def test_convert_str_to_devicetype_should_return_expected_devicetype(str, type):
     ])
 def test_convert_str_to_devicetype_with_unknown_device_should_return_mini(str, type):
     assert_that(tools.convert_str_to_devicetype(str)).is_equal_to(type)
+
+
+@mark.parametrize("username, password, token", [
+    ("test@switcher.com", "11223344", '{"token": "zvVvd7JxtN7CgvkD1Psujw=="}')
+    ])
+def test_get_token_should_return_valid_token(username, password, token):
+    assert_that(tools.get_token(username, password).to_json()).is_equal_to(token)
+
+
+@mark.parametrize("username, password, token", [
+    ("username", "password", '{"error": "Invalid credentials"}')
+    ])
+def test_get_token_with_error_should_return_empty_token(username, password, token):
+    assert_that(tools.get_token(username, password).to_json()).is_equal_to(token)
```

Comment:
I may be missing something, but does this test doing a real API call to Switcher servers? If it does, you should patch it not to do any externals IO/network calls.

Reply:
Yes actually yes.. they made this user and should stay for ever..
How can we simulate the api call otherwise?
its the get token function we are testing.

Reply:
Fixed to mock the request post itself.